tests/integration/targets/aci_fabric_pod_profile/tasks/main.yml

```diff
@@ -0,0 +1,122 @@
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: debug
+
+# CLEAN ENVIRONMENT
+- name: Ensure Fabric Pod Profile is absent before testing
+  aci_fabric_pod_profile:
+    <<: *aci_info
+    name: ansible_fabric_pod_profile
+    state: absent
+
+# CREATE FABRIC POD PROFILE
+- name: Create ansible_fabric_pod_profile
+  aci_fabric_pod_profile:
+    <<: *aci_info
+    name: ansible_fabric_pod_profile
+    description: Ansible test pod profile
+    state: present
+  register: add_fabric_pod_profile
+
+- name: Verify profile creation
+  assert:
+    that:
+    - add_fabric_pod_profile is changed
+    - add_fabric_pod_profile.current.0.fabricPodP.attributes.dn == "uni/fabric/podprof-ansible_fabric_pod_profile"
+    - add_fabric_pod_profile.current.0.fabricPodP.attributes.name == "ansible_fabric_pod_profile"
+    - add_fabric_pod_profile.current.0.fabricPodP.attributes.descr == "Ansible test pod profile"
+    - add_fabric_pod_profile.current.0.fabricPodP.attributes.annotation == 'orchestrator:ansible'
+
+# CREATE FABRIC POD PROFILE AGAIN TO TEST IDEMPOTENCE
+- name: Create ansible_fabric_pod_profile again
+  aci_fabric_pod_profile:
+    <<: *aci_info
+    name: ansible_fabric_pod_profile
+    description: Ansible test pod profile
+    state: present
+  register: add_fabric_pod_profile_again
+
+- name: Verify profile creation idempotence
+  assert:
+    that:
+    - add_fabric_pod_profile_again is not changed
+    - add_fabric_pod_profile_again.current.0.fabricPodP.attributes.dn == "uni/fabric/podprof-ansible_fabric_pod_profile"
+    - add_fabric_pod_profile_again.current.0.fabricPodP.attributes.name == "ansible_fabric_pod_profile"
+    - add_fabric_pod_profile_again.current.0.fabricPodP.attributes.descr  == "Ansible test pod profile"
+
+# QUERY FABRIC POD PROFILE
+- name: query fabric pod profile
+  aci_fabric_pod_profile:
+    <<: *aci_info
+    name: ansible_fabric_pod_profile
+    state: query
+  register: query_pod_profile
+
+- name: Verify profile query
+  assert:
+    that:
+    - query_pod_profile is not changed
+    - query_pod_profile.current.0.fabricPodP.attributes.dn == "uni/fabric/podprof-ansible_fabric_pod_profile"
+    - query_pod_profile.current.0.fabricPodP.attributes.name == "ansible_fabric_pod_profile"
+    - query_pod_profile.current.0.fabricPodP.attributes.descr == "Ansible test pod profile"
+
+# QUERY ALL POD PROFILES
+- name: query all fabric pod profiles
+  aci_fabric_pod_profile:
+    <<: *aci_info
+    state: query
+  register: query_pod_profile_all
+
+- name: Verify profile query idempotence
+  assert:
+    that:
+    - query_pod_profile_all is not changed
+
+# DELETE POD PROFILE
+- name: Delete ansible_fabric_pod_profile
```

Comment:
Please add test with checkmode and assert proposed and previous for the test.

Reply:
Done